sdk/identity/Azure.Identity/src/CliCredential.cs

```diff
@@ -0,0 +1,135 @@
+
+        private readonly CredentialPipeline _pipeline;
+        private readonly CliCredentialClient _client;
+
+        /// <summary>
+        /// Create an instance of CliCredential class.
+        /// </summary>
+        public CliCredential()
+            : this(CredentialPipeline.GetInstance(null), new CliCredentialClient())
+        { }
+
+        internal CliCredential(CredentialPipeline pipeline)
+            : this(pipeline, new CliCredentialClient())
+        { }
+
+        internal CliCredential(CredentialPipeline pipeline, CliCredentialClient client)
+        {
+            _pipeline = pipeline;
+
+            _client = client;
+        }
+
+        /// <summary>
+        /// Obtains a access token from Azure CLI credential, using this access token to authenticate. This method called by Azure SDK clients.
+        /// </summary>
+        /// <param name="requestContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken = default)
+        {
+            return GetTokenImplAsync(requestContext, cancellationToken).GetAwaiter().GetResult().GetTokenOrThrow();
+        }
+
+        /// <summary>
+        /// Obtains a access token from Azure CLI service, using the access token to authenticate. This method id called by Azure SDK clients.
+        /// </summary>
+        /// <param name="requestContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public override async ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken = default)
+        {
+            return (await GetTokenImplAsync(requestContext, cancellationToken).ConfigureAwait(false)).GetTokenOrThrow();
+        }
+
+        async ValueTask<ExtendedAccessToken> IExtendedTokenCredential.GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            return await GetTokenImplAsync(requestContext, cancellationToken).ConfigureAwait(false);
+        }
+
+        ExtendedAccessToken IExtendedTokenCredential.GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            return GetTokenImplAsync(requestContext, cancellationToken).GetAwaiter().GetResult();
+        }
+
+        private async ValueTask<ExtendedAccessToken> GetTokenImplAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            using CredentialDiagnosticScope scope = _pipeline.StartGetTokenScope("Azure.Identity.CliCredential.GetToken", requestContext);
+
+            try
+            {
+                string command = ScopeUtilities.ScopesToResource(requestContext.Scopes);
+                string extendCommand = $"az account get-access-token --resource {command}";
```

Comment:
We should explicitly pass `--output json` here, in case the user has a different default set.